examples/iPi/zundel/zundel_IP.ipynb

```diff
@@ -0,0 +1,777 @@
+    "from rascal.representations import SphericalInvariants\n",
+    "from rascal.neighbourlist import AtomsList\n",
+    "from rascal.utils import from_dict, to_dict, CURFilter, dump_obj, load_obj"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the snippets below we extract the relevant properties for each ASE frame. We load the total potential energies and we use the $\\bf{ASE.Atoms.arrays}$ methods to get the atomic forces. For more information on how to use ASE-related methods, check the documentation at https://wiki.fysik.dtu.dk/ase/ase/atoms.html.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dict_keys(['numbers', 'positions', 'forces'])\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Load the first N structures of the zundel dataset\n",
+    "N_dataset = 1000\n",
+    "frames = read('zundel_dataset.xyz', index=':{}'.format(N_dataset))\n",
+    "energies = np.loadtxt('zundel_energies.txt')[:N_dataset] \n",
+    "\n",
+    "#Keys of the arrays dictionary\n",
+    "print(frames[0].arrays.keys())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "init_cell": true
+   },
+   "outputs": [],
+   "source": [
+    "def extract_forces(frames,array_key='zeros'):\n",
+    "    f = []\n",
+    "    for frame in frames:\n",
+    "        if array_key is None:\n",
+    "            pass\n",
+    "        elif array_key == 'zeros':\n",
+    "            f.append(np.zeros(frame.get_positions().shape))\n",
+    "        else:\n",
+    "            f.append(frame.get_array(array_key))\n",
+    "    try:\n",
+    "        f = np.concatenate(f)\n",
+    "    except:\n",
+    "        pass\n",
+    "    return f"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build the GAP model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The preliminary step towards building a GAP model is defining a training and a test set. The training set is built by random selection of $800$ of the available structures, while the remaining $200$ will represent the test set. At this point, we also extract the distinct atomic species present in our system, define the atomic baseline energy and extract the information regarding energies and forces in the train and test set structures, which we will need later on."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "code_folding": []
+   },
+   "outputs": [],
+   "source": [
+    "# Number of structures to train the model with\n",
+    "n = 800\n",
+    "\n",
+    "global_species = []\n",
+    "for frame in frames:\n",
+    "    global_species.extend(frame.get_atomic_numbers())\n",
+    "global_species = np.unique(global_species)\n",
```

Comment:
nitpick: this can get quite inefficient, why not use a set?
```suggestion
    "global_species = set()\n",
    "for frame in frames:\n",
    "    global_species.add(frame.get_atomic_numbers())\n",
    "global_species = np.array(global_species)\n",
```